### PR TITLE
Send XML to workshops from Designer

### PR DESCRIPTION
**Pull request by Polej**

### CODE REVIEW

File: src/components/Pages/Admin/Designer.vue [hunk 1]
```diff
@@ -429,6 +448,61 @@ export default {
...
+            const groupsObjects = await this.findGroups({
+                query: {
+                    class: this.code,
+                },
+            });
+            groupsObjects.forEach((g) => { this.createOrUpdateWorkshopWithXML(g._id, xml); });
+        },
+
+        async createOrUpdateWorkshopWithXML(groupId, xml) {
+            let w;
```
Comment (IwoHerka) [inline src/components/Pages/Admin/Designer.vue]: Use descriptive names.

File: src/components/Pages/Admin/Designer.vue [hunk 2]
```diff
@@ -429,6 +448,61 @@ export default {
...
+            try {
+                await this.getWorkshopToStore(groupId);
+                w = await this.getWorkshop(groupId);
+            } catch (error) {
+                if (error.code !== 404) {
+                    this.alert = {
+                        type: 'danger',
+                        message: `Error @ get workshop: ${error.message}`,
```
Comment (IwoHerka) [inline src/components/Pages/Admin/Designer.vue]: This is not a message for end-user, is it? Rewrite it please to something user-friendy.

File: src/components/Pages/Admin/Designer.vue [hunk 3]
```diff
@@ -429,6 +448,61 @@ export default {
...
+            try {
+                if (w) {
+                    await this.updateWorkshop([groupId, { xml }]);
+                } else {
+                    await this.createWorkshop({ id: groupId, xml });
+                }
+            } catch (error) {
+                this.alert = {
+                    type: 'danger',
+                    message: `Error @ update/create workshop: ${error.message}`,
```
Comment (IwoHerka) [inline src/components/Pages/Admin/Designer.vue]: ^

File: src/components/Pages/Admin/Designer.vue [hunk 4]
```diff
@@ -429,6 +448,61 @@ export default {
             this.GI.setXML(freshGeogebraState);
             this.GI.registerGlobalListeners();
         },
+
+        async send(groups) {
```
Comment (IwoHerka) [inline src/components/Pages/Admin/Designer.vue:453]: Let his code breathe a little. Add new lines between blocks. This applies to the whole pull-request.

**Review by IwoHerka — CHANGES_REQUESTED** [review]

File: src/components/Pages/Admin/Designer.vue (hunk 1 above)
Comment (Polej) [reply src/components/Pages/Admin/Designer.vue]: Actually, what's wrong with this one, I think its quite descriptive.

**Pull request merged**